core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceSettingServiceUnitTests.scala

```diff
@@ -648,4 +665,148 @@ class WorkspaceSettingServiceUnitTests extends AnyFlatSpec with MockitoTestUtils
+    when(workspaceRepository.getWorkspace(workspaceName, None)).thenReturn(Future.successful(Option(workspace)))
+
+    val workspaceSettings = List(PubliclyReadableSetting(PubliclyReadableConfig(false)))
+    val workspaceSettingRepository = mock[WorkspaceSettingRepository]
+    when(workspaceSettingRepository.getWorkspaceSettings(workspaceId)).thenReturn(Future.successful(List.empty))
+    when(
+      workspaceSettingRepository.createWorkspaceSettingsRecords(workspaceId,
+                                                                workspaceSettings,
+                                                                defaultRequestContext.userInfo.userSubjectId
+      )
+    ).thenReturn(Future.successful(workspaceSettings))
+    workspaceSettings.foreach(workspaceSetting =>
+      when(workspaceSettingRepository.markWorkspaceSettingApplied(workspaceId, workspaceSetting.settingType))
+        .thenReturn(Future.successful(1))
+    )
+
+    val samDAO = mock[SamDAO](RETURNS_SMART_NULLS)
+    when(samDAO.getUserStatus(any()))
+      .thenReturn(Future.successful(Option(SamUserStatusResponse("fake_user_id", "user@example.com", true))))
+    when(
+      samDAO.userHasAction(ArgumentMatchers.eq(SamResourceTypeNames.workspace),
+                           ArgumentMatchers.eq(workspaceId.toString),
+                           ArgumentMatchers.eq(SamWorkspaceActions.own),
+                           any()
+      )
+    ).thenReturn(Future.successful(true))
+    when(
+      samDAO.setPolicyPublic(
+        ArgumentMatchers.eq(SamResourceTypeNames.workspace),
+        ArgumentMatchers.eq(workspaceId.toString),
+        ArgumentMatchers.eq(SamWorkspacePolicyNames.reader),
+        ArgumentMatchers.eq(false),
+        any()
+      )
+    ).thenReturn(Future.successful(()))
+    when(samDAO.getAllUsersGroup(any())).thenReturn(Future.successful(WorkbenchEmail("allUsers@fc.org")))
+
+    val gcsDAO = mock[GoogleServicesDAO](RETURNS_SMART_NULLS)
+    when(gcsDAO.terraBucketReaderRole).thenReturn("terra-bucket-reader")
+
+    val googleStorageService = mock[GoogleStorageService[IO]](RETURNS_SMART_NULLS)
+    when(
+      googleStorageService.removeIamPolicy(
+        ArgumentMatchers.eq(GcsBucketName(workspace.bucketName)),
+        ArgumentMatchers.eq(
+          Map[StorageRole, NonEmptyList[Identity]](
+            StorageRole.CustomStorageRole("terra-bucket-reader") -> NonEmptyList.one(Identity.group("allUsers@fc.org"))
+          )
+        ),
+        any(),
+        any(),
+        any()
+      )
+    ).thenReturn(fs2.Stream.empty)
+
+    val service =
+      workspaceSettingServiceConstructor(
+        samDAO = samDAO,
+        workspaceRepository = workspaceRepository,
+        workspaceSettingRepository = workspaceSettingRepository,
+        gcsDAO = gcsDAO,
+        googleStorageService = googleStorageService
+      )
+
+    Await.result(service.setWorkspaceSettings(workspaceName, workspaceSettings), Duration.Inf)
```

Comment:
for both of these tests, add some `verify` calls to make sure you actually invoked the policy calls to Sam and the IAM calls to the storage service?